File: Exercise/LuoGu/Code_2023_7_26/P1320.cpp
#include <iostream>
#include <algorithm>
#include <math.h>
#include <vector>
#include <string>

using namespace std;

int main(void)
{
    int digit, count = 0;
    bool is_one = false, has_show = false;
    string num_str;

    while (cin >> num_str)
    {
        if (!has_show)
        {
            cout << num_str.length() << " ";
            has_show = true;
        }
        for (int i = 0; i < num_str.length(); i++)
        {
            digit = num_str[i] - '0';
            if (digit == 1)
            {
                if (!is_one)
                {
                    is_one = true;
                    cout << count << " ";
                    count = 1;
                }
                else
                {
                    count++;
                }
            }
            else
            {
                if (is_one)
                {
                    is_one = false;
                    cout << count << " ";
                    count = 1;
                }
                else
                {
                    count++;
                }
            }
        }
    }
    cout << count << endl;

    return 0;
}

File: Exercise/LuoGu/Code_2023_7_26/P1320.java
package Code_2023_7_26;

import java.util.Scanner;

public class P1320 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String numStr;
        int count = 0, digit;
        boolean hasShow = false, isOne = false;
        while (sc.hasNextLine()) {
            numStr = sc.nextLine();
            if (!hasShow) {
                System.out.print(numStr.length() + " ");
                hasShow = true;
            }

            for (int i = 0; i < numStr.length(); i++) {
                digit = numStr.charAt(i) - '0';
                if (digit == 1) {
                    if (!isOne) {
                        isOne = true;
                        System.out.print(count + " ");
                        count = 1;
                    } else {
                        count++;
                    }
                } else {
                    if (isOne) {
                        isOne = false;
                        System.out.print(count + " ");
                        count = 1;
                    } else {
                        count++;
                    }
                }
            }
        }
        System.out.print(count + " ");

        sc.close();
    }
}
